### PR TITLE
Kev/102_change_key

### PR DESCRIPTION
**Pull request by junhaow1**
# Pull Request Details

## What issue does this PR address

- implemented change key widget in keypod app (frontend)

- Link to associated issue: https://github.com/anusii/solidpod/issues/102

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [ ] Pre-exisiting lint errors noted: [HERE]
- [x] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome on Android
  - [x] Chrome
  - [ ] iOS
  - [ ] Linux
  - [ ] MacOS
  - [ ] Windows
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

## Finalising

Once PR discussion is complete and 2 reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev

### CODE REVIEW

**Comment by cdawei** [comment]
I can see the button after logging in
button
But the background looks odd after clicking the "Change Encryption Key" button (is it transparent?)
background
This is a test of the Linux app. The chrome build (on Linux) shows the same for a short time then the background changes to white (but I can clearly see the flashing).

File: lib/screens/home.dart [hunk 1]
```diff
@@ -35,10 +35,12 @@
 library;
 
 import 'package:flutter/material.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
 
 import 'package:intl/intl.dart';
 import 'package:solidpod/solidpod.dart';
-
+import 'package:solidpod/src/widgets/setting.dart';
```
Comment (gjwgit) [inline lib/screens/home.dart:42]: Do not reference src files from a package.

These should be exported from solidpod.dar!

Also they are in the wrong ordering here according to the style guide.

Comment (gjwgit) [inline lib/screens/home.dart:42]: I think this (setting.dart) should become a popup widget. Then all of the code here should be replaced by a button "Change Key" which calls SolidKeyChange() to popup a little window with the interface you have now (old, new, repeat text entry and two buttons CHANGE and CANCEL).

**Review by gjwgit — CHANGES_REQUESTED** [review]

File: lib/screens/home.dart (hunk 1 above)
Comment (junhaow1) [reply lib/screens/home.dart:42]: > I think this (setting.dart) should become a popup widget. Then all of the code here should be replaced by a button "Change Key" which calls SolidKeyChange() to popup a little window with the interface you have now (old, new, repeat text entry and two buttons CHANGE and CANCEL).

got it, working on it now :)

**Review by gjwgit — CHANGES_REQUESTED** [review]
Add a new button to the current app home page to call ChangeSolidKeyDialog() as the demo.

File: lib/main.dart [hunk 2]
```diff
@@ -125,7 +125,10 @@ class KeyPod extends StatelessWidget {
         loginButtonStyle: LoginButtonStyle(
           background: Colors.green,
         ),
-        child: Home(),
+
+        child: Home(
+            // appName: 'KeyPod',
+            ),
```
Comment (cdawei) [inline lib/main.dart:131]: This change is unnecessary.

File: lib/screens/home.dart [hunk 3]
```diff
@@ -37,6 +37,7 @@ library;
 import 'package:flutter/material.dart';
 
 import 'package:intl/intl.dart';
+import 'package:solidpod/solidpod.dart';
```
Comment (cdawei) [inline lib/screens/home.dart:40]: Why importing solidpod twice?

File: pubspec.yaml [hunk 4]
```diff
@@ -19,6 +19,7 @@ dependencies:
   flutter:
     sdk: flutter
   flutter_launcher_icons: ^0.13.1
+  flutter_secure_storage: ^9.0.0
```
Comment (cdawei) [inline pubspec.yaml:22]: Where do we need flutter secure storage in keypod?

**Review by cdawei — CHANGES_REQUESTED** [review]
Hi Kevin, thanks for the changes, the pop-up worked in my test. Please fix/clarify the minor issues in comments.

File: lib/main.dart (hunk 2 above)
Comment (junhaow1) [reply lib/main.dart:131]: agree, done

File: pubspec.yaml (hunk 4 above)
Comment (junhaow1) [reply pubspec.yaml:22]: removed

File: lib/screens/home.dart (hunk 3 above)
Comment (junhaow1) [reply lib/screens/home.dart:40]: removed

**Review by cdawei — APPROVED** [review]

**Review by gjwgit — APPROVED** [review]
Not fully tested but let's get it in to dev asap.

**Pull request merged**